### PR TITLE
NIFI-12016: Allow use of compatible NAR bundles when loading flow from cluster connection (backport)

### PR DESCRIPTION
**Pull request by mattyb149**
# Summary

[NIFI-12016](https://issues.apache.org/jira/browse/NIFI-12016) When determining what bundles are compatible, consider not just any bundle if it's the only one but also any bundle whose version matches the framework version so that when NiFi is upgraded, it is handled more gracefully.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `support/nifi-1.x` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/util/BundleUtils.java [hunk 1]
```diff
@@ -30,11 +31,29 @@
...
+        // During an upgrade, it's fairly common to have two versions of a NAR: the version shipped with NiFi and another version, perhaps to maintain
+        // backward compatibility to because the new version behaves some different way and the user wants the old behavior in some instances, etc.
+        // In this case, the user may have two versions. For example, version 2.2.0 and 2.4.0 while NiFi is at version 2.4.0.
+        // Now, during upgrade to 2.4.1, there will no longer be a 2.4.0 available. We want to be smart enough to realize that those extension using version
+        // 2.2.0 stay there but those using 2.4.0 upgrade to 2.4.1.
+        // To do this, we always first match on the exact version but this method is called when there's no exact match. So those marked 2.2.0 won't arrive here.
+        // But for those extensions that were using 2.4.0, we want to now look for version 2.4.1 - I.e., the one with the same version as the framework. If we
+        // find that version, then we want to use it. This helps to smooth out the upgrade process even when users have multiple versions of a given NAR.
```
Comment (dan-s1) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/util/BundleUtils.java:48]: These comments reference 2.x versions. They should be updated to indicate 1.x as this fix is for the 1.x branch

**Comment by mosermw** [comment]
@mattyb149 Is this is still wanted for the 1.x branch?

**Comment by mattyb149** [comment]
Yes please :)

**Comment by mosermw** [comment]
+1 merging

**Pull request closed without merging**